### PR TITLE
Chore: Migrate `gf-form` classes to emotion globals

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- migrates `gf-form` classes to emotion globals

**Why do we need this feature?**

- clean up old deprecated SCSS styles 🧹 

**Who is this feature for?**

- grafana devs

**Which issue(s) does this PR fix?**:

For https://github.com/grafana/grafana/issues/64129

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/sass/_angular.scss [hunk 1]
```diff
@@ -2171,3 +2172,51 @@ $easing: cubic-bezier(0, 0, 0.265, 1);
 div.flot-text {
   color: $text-color !important;
 }
+
+.gf-form-dropdown-typeahead {
```
Comment (tskarhed) [inline public/sass/_angular.scss:2176]: Why can't this one be moved to GlobalStyles or vice versa?

Comment (ashharrison90) [reply public/sass/_angular.scss:2176]: it can, but there's no need since no one is using it. might as well remove it completely 👍

**Review by JoaoSilvaGrafana — APPROVED** [review]

File: packages/grafana-ui/src/themes/GlobalStyles/forms.ts [hunk 2]
```diff
@@ -27,5 +27,345 @@ export function getFormElementStyles(theme: GrafanaTheme2) {
...
+      lineHeight: '32px',
+      marginRight: theme.spacing(0.5),
+      borderRadius: theme.shape.radius.default,
+      justifyContent: 'space-between',
+      border: 'none',
+
+      '&--grow': {
+        flexGrow: 1,
+      },
+
```
Comment (eledobleefe) [inline packages/grafana-ui/src/themes/GlobalStyles/forms.ts:154]: Don't we have to transform the following one?
```jsx
 &--error {
    color: $critical;
  }
```

Comment (eledobleefe) [reply packages/grafana-ui/src/themes/GlobalStyles/forms.ts:154]: > Don't we have to transform the following one?
> 
> ```js
>  &--error {
>     color: $critical;
>   }
> ```

The missing ones are not used and that is why you got rid of them, no?

Comment (ashharrison90) [reply packages/grafana-ui/src/themes/GlobalStyles/forms.ts:154]: yep exactly. no one's using them - time to get rid 🧹

**Pull request merged**